src/standalone/Soil/Biogeochemistry/Biogeochemistry.jl

```diff
@@ -435,20 +610,49 @@ This has been written so as to work with Differential Equations.jl.
 """
 function ClimaLand.make_update_aux(model::SoilCO2Model)
     function update_aux!(p, Y, t)
+        # Enforce positivity of prognostic variables to prevent unphysical values
+        # This acts as a safety net after the ODE integrator updates Y
+        FT = eltype(Y.soilco2.C)
+        Y.soilco2.C .= max.(Y.soilco2.C, FT(0))
+        Y.soilco2.O2_a .= max.(Y.soilco2.O2_a, FT(0))
+        Y.soilco2.SOC .= max.(Y.soilco2.SOC, FT(0))
+
         params = model.parameters
         z = model.domain.fields.z
         T_soil = soil_temperature(model.drivers.met, p, Y, t, z)
         θ_l = soil_moisture(model.drivers.met, p, Y, t, z)
-        Csom = p.drivers.soc
+        Csom = Y.soilco2.SOC  # Now using prognostic SOC
         P_sfc = p.drivers.P
         θ_w = θ_l
         ν = model.drivers.met.ν
         θ_a100 = model.drivers.met.θ_a100
         b = model.drivers.met.b
 
+        # Store temperature in auxiliary variables (works for both standalone and coupled modes)
+        @. p.soilco2.T = T_soil
+
         p.soilco2.D .=
             co2_diffusivity.(T_soil, θ_w, P_sfc, θ_a100, b, ν, params)
-        p.soilco2.Sm .= microbe_source.(T_soil, θ_l, Csom, ν, params)
+        # O2 diffusivity is the same as CO2 diffusivity
+        p.soilco2.D_o2 .=
+            co2_diffusivity.(T_soil, θ_w, P_sfc, θ_a100, b, ν, params)
+
+        # Compute volumetric air content and store in auxiliary variables
+        FT = eltype(Y.soilco2.C)
+        p.soilco2.θ_a .= @. max(ν - θ_l, FT(0))
+
+        # Compute O2 mass concentration (kg O2/m³ air) from O2_a for diffusion (ideal gas law)
+        @. p.soilco2.O2 =
+            o2_concentration(Y.soilco2.O2_a, T_soil, P_sfc, params)
+
+        # Compute O2 availability (dimensionless) for microbial kinetics (tortuosity-based)
+        (; D_oa) = params
+        @. p.soilco2.O2_avail =
+            o2_availability(Y.soilco2.O2_a, p.soilco2.θ_a, D_oa)
+
+        # Compute microbial source using O2 availability
+        p.soilco2.Sm .=
```

Comment:
```suggestion
        @. p.soilco2.Sm =microbe_source(T_soil, θ_l, Csom, p.soilco2.O2_avail, params)
```

and elsewhere...

this saves us a kernel launch (every .=, .*, etc is a kernel launch, the @. fuses them into a single one.